tf2_ros/src/transform_listener.cpp

```diff
@@ -57,7 +57,6 @@ TransformListener::~TransformListener()
   using_dedicated_thread_ = false;
   if (dedicated_listener_thread_) {
     dedicated_listener_thread_->join();
-    delete dedicated_listener_thread_;
```

Comment:
couldn't the complete destructor be placed in the deleter of the unique_ptr? The `if` branch could be completely eliminated because the unique_ptr would be a nullptr if no dedicated thread is started.

Reply:
@Karsten1987 that's a good point. I'll definitely do that.